decidim-core/db/migrate/20250217192438_convert_user_groups_into_users.rb

```diff
@@ -21,6 +21,15 @@ class UserGroup < ApplicationRecord
   # rubocop:disable Rails/SkipsModelValidations
   def up
     User.old_group.find_each do |group|
+      if group.email.blank? || User.where.not(id: group.id).exists?(decidim_organization_id: group.decidim_organization_id, email: group.email)
```

Comment:
Let's try to be more clear on what is happening here. Let's extract these two conditions to a method so they can better explain what is happening. The first it may be too much, but I think the second should be there. Feel free to rename the methods if you feel so. 

As an alternative we can add a comment, but I prefer to make the code clearer if possible

```suggestion
      if empty_email?(group) || user_group_with_a_duplicated_email_from_a_user?(group)
```

Reply:
Fixed in [a0f5ef3](https://github.com/decidim/decidim/pull/15289/commits/a0f5ef3516ad859db689057290b25b9d6fb66569)